pkg/kubelego/watch.go

```
@@ -49,10 +61,44 @@ func (kl *KubeLego) WatchReconfigure() {
 			if quit {
 				return
 			}
-			kl.Log().Debugf("worker: begin processing %v", item)
-			kl.Reconfigure()
-			kl.Log().Debugf("worker: done processing %v", item)
-			kl.workQueue.Done(item)
+			func(item interface{}) {
+				defer kl.workQueue.Done(item)
+				key, ok := item.(string)
+				if !ok {
+					kl.Log().Errorf("worker: invalid item in workqueue: %v", item)
+					kl.workQueue.Forget(item)
+					return
+				}
+				name, namespace, err := cache.SplitMetaNamespaceKey(key)
+				if err != nil {
+					kl.Log().Errorf("worker: invalid string in workqueue: %s", item)
+					kl.workQueue.Forget(item)
+					return
+				}
+				kl.Log().Debugf("worker: begin processing %v", key)
+				// attempt to get an internal ingress type.
+				ing := ingress.New(kl, namespace, name)
+				// if it doesn't exist for some reason, exit here and forget the
+				// item from the workqueue.
+				if ing.Exists == false {
+					kl.Log().Errorf("worker: ingress for key %q no longer exists. Skipping...", key)
+					kl.workQueue.Forget(item)
+					return
+				}
+				// attempt to process the ingress
+				err = kl.reconfigure(ing)
+				if err != nil {
+					kl.Log().Errorf("worker: error processing item, requeuing after rate limit: %v", err)
+					// we requeue the item and skip calling Forget here to ensure
+					// a rate limit is applied when adding the item after a failure
+					kl.workQueue.AddRateLimited(key)
+					return
+				}
```

Yeah , it looks like we should return the errors from `reconfigure`.